src/MyConnection.h

```diff
@@ -18,7 +18,8 @@ class MyConnection : boost::noncopyable {
 
 public:
 
-  MyConnection(std::string host, std::string user, std::string password,
+  MyConnection(std::string host, std::string user,
+               Rcpp::Nullable < std::string > password,
```

Comment:
For password, there's a difference between "" (empty password) and NULL (no password). Not for the other options.